config/main.py

```diff
@@ -542,32 +587,96 @@ def config():
-@click.argument('filename', default='/etc/sonic/config_db.json', type=click.Path(exists=True))
-def load(filename, yes):
+@click.argument('filename', default=DEFAULT_CONFIG_DB_FILE, type=click.Path(exists=True))
+def load(filename, yes, namespace):
     """Import a previous saved config DB dump file."""
     if not yes:
         click.confirm('Load config from the file %s?' % filename, abort=True)
-    command = "{} -j {} --write-to-db".format(SONIC_CFGGEN_PATH, filename)
-    run_command(command, display_cmd=True)
+
+    if namespace:
+        if not is_multi_asic():
+            click.echo("namespace is not significant in a Single ASIC device")
+            return None
+        if not validate_namespace(namespace):
+            click.echo("invalid Namespace entered {}".format(namespace))
+            return None
+        if filename == DEFAULT_CONFIG_DB_FILE:
+            inst = namespace[len(NS_PREFIX)]
+            filename = "/etc/sonic/config_db{}.json".format(inst)
+            click.echo("valid config_db file for namespace {} would be {}".format(namespace, filename))
+            return None
+
+        command = "{} -n {} -j {} --write-to-db".format(SONIC_CFGGEN_PATH, namespace, filename)
+        run_command(command, display_cmd=True)
+    else:
+        # Load config for the database service running on linux host
+        command = "{} -j {} --write-to-db".format(SONIC_CFGGEN_PATH, filename)
+        run_command(command, display_cmd=True)
+
+        """In case of multi-asic mode we have additional config_db{NS}.json files for
+           various namespaces created per ASIC. {NS} is the namespace index.
+        """
+        if is_multi_asic():
+            ns_list = get_all_namespaces()
+            namespaces = ns_list['front_ns'] + ns_list['back_ns']
+            for namespace in namespaces:
+                inst = namespace[len(NS_PREFIX)]
+                filename = "/etc/sonic/config_db{}.json".format(inst)
+                if os.path.isfile(filename):
+                    command = "{} -n {} -j {} --write-to-db".format(SONIC_CFGGEN_PATH, namespace, filename)
+                    run_command(command, display_cmd=True)
+                else:
+                    click.echo("The config_db file {} doesn't exist".format(filename))
 
 @config.command()
 @click.option('-y', '--yes', is_flag=True)
 @click.option('-l', '--load-sysinfo', is_flag=True, help='load system default information (mac, portmap etc) first.')
-@click.argument('filename', default='/etc/sonic/config_db.json', type=click.Path(exists=True))
+@click.argument('filename', default=DEFAULT_CONFIG_DB_FILE, type=click.Path(exists=True))
 def reload(filename, yes, load_sysinfo):
     """Clear current configuration and import a previous saved config DB dump file."""
     if not yes:
-        click.confirm('Clear current config and reload config from the file %s?' % filename, abort=True)
+        click.confirm('Clear current config and reload config ?', abort=True)
```

Comment:
So, for multi-ASIC platforms, there is no way to specify a file to load from because there are multiple files involved, is this correct? If so, maybe we should consider fixing the file for single-ASIC platforms as well, so that the behavior is consistent for both.

Reply:
Yes there are multiple files that need to be specified, might be difficult for user also to give as input. Are you referring to removing the filename option itself from the command ?  if we remove won't it affect backward compatibility with some scripts using this command may be ?

Reply:
Yes, we would need to check. Most of the time we just load from the default conifg_db.json file.

Reply:
I am finding couple of testcases which uses filename option to specify the config_db.json file, there could be other use cases as well. 
Hence I have updated the implementation so that the user can input specific config_db.json files even for namespace. This will make sure the behavior is consistent for all platforms. Please take a look.